packages/smithy-aws-core/src/smithy_aws_core/auth/sigv4.py

```diff
@@ -0,0 +1,54 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  SPDX-License-Identifier: Apache-2.0
+from dataclasses import dataclass
+from typing import Protocol
+
+from smithy_aws_core.identity import AWSCredentialsIdentity
+from smithy_core.aio.interfaces.identity import IdentityResolver
+from smithy_core.exceptions import SmithyIdentityException
+from smithy_core.interfaces.identity import IdentityProperties
+from smithy_http.aio.interfaces.auth import HTTPAuthScheme, HTTPSigner
+from aws_sdk_signers import SigV4SigningProperties, SigV4Signer
+
+
+class SigV4Config(Protocol):
+    aws_credentials_identity_resolver: (
+        IdentityResolver[AWSCredentialsIdentity, IdentityProperties] | None
+    )
+
+
+@dataclass(init=False)
+class SigV4AuthScheme(
+    HTTPAuthScheme[
+        AWSCredentialsIdentity, SigV4Config, IdentityProperties, SigV4SigningProperties
+    ]
+):
+    """SigV4 AuthScheme."""
+
+    scheme_id: str
+    signer: HTTPSigner[AWSCredentialsIdentity, SigV4SigningProperties]
+
+    def __init__(
+        self,
+        *,
+        signer: HTTPSigner[AWSCredentialsIdentity, SigV4SigningProperties]
+        | None = None,
+    ) -> None:
+        """Constructor.
+
+        :param identity_resolver: The identity resolver to extract the api key identity.
+        :param signer: The signer used to sign the request.
+        """
+        self.scheme_id = "aws.auth#sigv4"
+        # TODO: There are type mismatches in the signature of the "sign" method.
```

Comment:
Do you know what the mismatches were? I can take a look at those?

Reply:
I think its primarily a naming, http_request vs request, see: https://github.com/smithy-lang/smithy-python/blob/develop/packages/smithy-http/src/smithy_http/aio/interfaces/auth.py#L39